### ISSUE TITLE
CatalogName & catelogType for Identity service set to false

### ISSUE THREAD

**Issue by haphan** [php-opencloud/openstack#54]
@jamiehannaford Any reason why these params are set to false by default?

https://github.com/php-opencloud/openstack/blob/master/src/OpenStack.php#L134

Are you expecting user to set it manually like this?
```php 
$identity = $openstack->identityV3(
    [
        'region'      => '{region}',
        'catalogName' => 'keystone',
        'catalogType' => 'identity',
    ]
);
```

**Comment by jamiehannaford** [comment]
@haphan Well, the service builder doesn't pass on those values, because they're only used for authentication (i.e. extracting service URL and token ID from the catalog):

https://github.com/php-opencloud/common/blob/master/src/Common/Service/Builder.php#L95
https://github.com/php-opencloud/openstack/blob/master/src/Identity/v3/Service.php#L36-L37

We can add them anyway, but I can't see how those strings will ever be used.

**Comment by haphan** [comment]
@jamiehannaford 
Because this 
```php
public function getServiceUrl(string $name, string $type, string $region, string $urlType): string
````
will throw an error when `$name` and `$type` are `null` in
https://github.com/php-opencloud/openstack/blob/master/src/Identity/v3/Models/Catalog.php#L34

**Comment by jamiehannaford** [comment]
@haphan I'm not sure I understand the problem. The identity service never calls `authenticate`, it has a user-defined service URL. Otherwise there would be an infinite loop

**Comment by haphan** [comment]
@jamiehannaford Identity service does not directly call `authenticate` but the `$options['authHandler']` does which basically the same thing.

The implementation of `authenticate` in V3 is slightly different from that of V2 which has `$serviceUrl`.

To reproduce the exception, try this
```php
<?php

require 'vendor/autoload.php';

$params = [
    'authUrl' => 'http://1.2.3.4:5000/v3',
    'region'  => '{region}',
    'user'    =>
        [
            'name'     => 'fooo',
            'password' => 'barrr',
            'domain'   =>
                [
                    'id' => 'default',
                ],
        ],

];
$openstack = new OpenStack\OpenStack($params);
$identity = $openstack->identityV3(['region' => '{region}']);
print_r(iterator_to_array($identity->listProjects()));
```

**Comment by jamiehannaford** [comment]
@haphan Ah, I see what you mean. The bool types are throwing an `Uncaught TypeError`. Yeah, completely happy to change those `false` vals to strings. Great catch!

**Comment by jamiehannaford** [comment]
Fixed in #55
